content/en/blog/_posts/2023-09-20-kubeadm-use-etcd-learner-mode.md

```diff
@@ -0,0 +1,81 @@
+---
+
+**Author:** Paco Xu(DaoCloud)
+
+Kubeadm now supports ETCD learner mode, which allows you to enhance the resilience and stability
+of your Kubernetes clusters by leveraging the [learner mode](https://etcd.io/docs/v3.4/learning/design-learner/#appendix-learner-implementation-in-v34) feature introduced in ETCD version 3.4.
+This guide will walk you through using ETCD learner mode with kubeadm.
+
+In v1.27, kubeadm introduced a new feature gate `EtcdLearnerMode`. With this feature gate enabled,
+when joining a new control plane node, a new etcd member will be created as a learner and 
+promoted to a voting member only after the etcd data are fully aligned.
+
+## What are the advantages of using ETCD learner mode?
+
+Etcd learner mode offers several compelling reasons to consider its adoption
+in Kubernetes clusters:
+
+1. **Enhanced Resilience**: Etcd learner nodes are non-voting members that catch up with
+the leader's logs before becoming fully operational. This prevents new cluster members
+from disrupting the quorum or causing leader elections, making the cluster more resilient
+during membership changes.
+2. **Reduced Cluster Unavailability**: Traditional approaches to adding new members often
+result in cluster unavailability periods, especially in slow infrastructure or misconfigurations.
+Etcd learner mode minimizes such disruptions.
+3. **Simplified Maintenance**: Learner nodes provide a safer and reversible way to add or replace
+cluster members. This reduces the risk of accidental cluster outages due to misconfigurations or
+missteps during member additions.
+4. **Improved Network Tolerance**: In scenarios involving network partitions, learner mode allows
+for more graceful handling. Depending on the partition a new member lands, it can seamlessly
+integrate with the existing cluster without causing disruptions.
+
+In summary, the etcd learner mode improves the reliability and manageability of Kubernetes clusters
+during member additions and changes, making it a valuable feature for cluster operators.
+
+## How to join the control plane using the new mode?
+
+### Create an ETCD Learner Mode Cluster
+
+A general process for creating Highly Available Clusters with kubeadm can refer to 
+[configure and deploy an HA control plane](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/high-availability/).
+
+To create a Kubernetes cluster with ETCD learner mode using kubeadm, follow these steps:
+
+```shell
+# kubeadm init --feature-gates=EtcdLearnerMode=true ...
+kubeadm init --config=kubeadm-config.yaml
+```
+
+The kubeadm configuration file is like below:
+
+```yaml
+apiVersion: kubeadm.k8s.io/v1beta3
+kind: ClusterConfiguration
+featureGates:
+  EtcdLearnerMode: true
+```
+
+The ETCD learner mode will be enabled during cluster creation.
+
+### Join Nodes to the ETCD Learner Mode Cluster
+
+Before joining a node to the ETCD learner mode cluster, ensure that the control plane nodes and
```

Comment:
```suggestion
Before joining a worker node to the ETCD learner mode cluster, ensure that the control plane nodes and
```
?

Reply:
Before joining a control-plane node to the etcd learner mode cluster, ensure that the control plane nodes
and all etcd members are healthy.

Before joining a worker node to the etcd learner mode cluster, ensure that the control plane nodes are healthy.